### PR TITLE
50534 Left hand menu & tabs available on error page

### PR DESCRIPTION
**Pull request by mohamme15315**
Coding
Unit Testing

### CODE REVIEW

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/NoticestoMariners/Index.cshtml [hunk 1]
```diff
@@ -12,50 +12,55 @@
...
+                        { ViewBag.TableTitle = "Partner"; }
+                        <partial name="ShowWeeklyFilesList" model="@partnerList" />
                         { ViewBag.TableTitle = "Public"; }
                     }
-                        <partial name="ShowWeeklyFilesList" model="@publicList" />
+                    <partial name="ShowWeeklyFilesList" model="@publicList" />
+                }
+                else
+                {
+                    <partial name="_error" />
```
Comment (usha13529) [inline UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/NoticestoMariners/Index.cshtml:63]:
```suggestion
                    <partial name="_Error" />
```

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/Home/Error.cshtml [hunk 2]
```diff
@@ -1,10 +1,37 @@
 ﻿@{
     ViewData["Title"] = "Maritime Safety Information";
+    Layout = "~/Views/Shared/_MSILayout.cshtml";
+}
+@section Scripts
+{
+<script type="text/javascript" src="@Url.Content("~/js/Custom/RadioNavigationalWarnings.js")"></script>
 }
```
Comment (usha13529) [inline UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/Home/Error.cshtml:8]: I believe there is no need to include RadioNavigationalWarnings.js on Error page. Is there any specific reason behind this?

**Review by usha13529 — CHANGES_REQUESTED** [review]
Few review comments to address please.

File: Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Controllers/HomeControllerTest.cs [hunk 3]
```diff
@@ -1,10 +1,17 @@
...
 using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
 using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Controllers;
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
+using Microsoft.Identity.Client;
 using NUnit.Framework;
-using System.Threading.Tasks;
 using UKHO.MaritimeSafetyInformation.Common.Configuration;
```
[the remark this answers is not in the document]
Comment (mohamme15315) [reply Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Controllers/HomeControllerTest.cs:15]: Done

File: Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Services/NMDataServiceTest.cs [hunk 4]
```diff
@@ -411,7 +413,7 @@ public async Task WhenGetWeeklyFilesResponseModelsAsyncWithZeroIsCalled_ThenShou
         }
 
         [Test]
-        public void WhenGetWeeklyFilesResponseModelsAsyncWithZeroIsCalled_ThenShouldThrowInvalidDataException()
+        public async Task WhenGetWeeklyFilesResponseModelsAsyncWithZeroIsCalled_ThenShouldRetunEmptyShowFilesResponseList()
```
[the remark this answers is not in the document]
Comment (mohamme15315) [reply Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Services/NMDataServiceTest.cs]: Done

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common/Logging/EventIds.cs [hunk 5]
```diff
@@ -656,7 +656,23 @@ public enum EventIds
         /// <summary>
         /// 910158 - Request for searching annual batch files from cache azure table storage is completed.
         /// </summary>
-        FSSSearchAnnualBatchFilesResponseStoreToCacheCompleted = 910158
+        FSSSearchAnnualBatchFilesResponseStoreToCacheCompleted = 910158,
+		/// <summary>
```
[the remark this answers is not in the document]
Comment (mohamme15315) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common/Logging/EventIds.cs]: Done

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Controllers/NoticestoMarinersController.cs [hunk 6]
```diff
@@ -29,13 +29,21 @@ public async Task<IActionResult> Index()
...
 
                 ShowWeeklyFilesResponseModel showWeeklyFiles = await _nMDataService.GetWeeklyFilesResponseModelsAsync(0, 0, GetCurrentCorrelationId());
 
                 _logger.LogInformation(EventIds.ShowWeeklyFilesResponseIndexGetCompleted.ToEventId(), "Maritime safety information request for weekly NM file response for index get completed for User:{SignInName} and IsDistributor:{IsDistributorUser} and correlationId:{correlationId}", _userService.SignInName ?? "Public", _userService.IsDistributorUser, GetCurrentCorrelationId());
 
+                if (showWeeklyFiles.ShowFilesResponseList.Count == 0)
+                {
+                    ViewBag.HasError = true;
+                    string correlationId = GetCurrentCorrelationId();
+                    ViewData["CurrentCorrelationId"] = correlationId;
```
[the remark this answers is not in the document]
Comment (mohamme15315) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Controllers/NoticestoMarinersController.cs]: Done

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Controllers/NoticestoMarinersController.cs [hunk 7]
```diff
@@ -53,12 +61,21 @@ public async Task<IActionResult> Index(int year, int week)
...
                 ViewData["Year"] = year;
                 ViewData["Week"] = week;
 
                 _logger.LogInformation(EventIds.ShowWeeklyFilesResponseIndexPostCompleted.ToEventId(), "Maritime safety information request for weekly NM file response for index post completed for User:{SignInName} and IsDistributor:{IsDistributorUser} with correlationId:{correlationId}", _userService.SignInName ?? "Public", _userService.IsDistributorUser, GetCurrentCorrelationId());
+
+                if (showWeeklyFiles.ShowFilesResponseList.Count == 0)
+                {
+                    ViewBag.HasError = true;
+                    string correlationId = GetCurrentCorrelationId();
+                    ViewData["CurrentCorrelationId"] = correlationId;
```
[the remark this answers is not in the document]
Comment (mohamme15315) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Controllers/NoticestoMarinersController.cs]: Done

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common/Models/NoticesToMariners/ShowNMFilesResponseModel.cs [hunk 8]
```diff
@@ -5,5 +5,7 @@ public class ShowNMFilesResponseModel
     {
         public List<ShowFilesResponseModel> ShowFilesResponseModel { get; set; }
         public bool IsBatchResponseCached { get; set; }
+
+        public static implicit operator List<object>(ShowNMFilesResponseModel v) => throw new NotImplementedException();
```
[the remark this answers is not in the document]
Comment (mohamme15315) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common/Models/NoticesToMariners/ShowNMFilesResponseModel.cs]: Done

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/NoticestoMariners/Index.cshtml (hunk 1 above)
Comment (mohamme15315) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/NoticestoMariners/Index.cshtml]: Done

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/Home/Error.cshtml (hunk 2 above)
Comment (mohamme15315) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/Home/Error.cshtml:8]: It is used to handle the title if any global exception occurs

**Review by usha13529 — APPROVED** [review]
Thanks @mohamme15315 , @Avinash15515 , @Pradeep15558 & @pradnya13705 for changes. Looks good to me.

**Review by Chownder — CHANGES_REQUESTED** [review]
It all looks to be functionally there, and I like the mechanism for doing this, but there are a couple of cases where we are showing descriptive text where we should just show the error.

Across the "Notices to Mariners" tabs, they should probably all have the form, as it would be more consistent and the information on the Cumulative and Annual pages are superfluous during an error.   

I was going to say similar for Leisure, but there's no consistency gain there, and that text has reference to handy resources.

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/NoticestoMariners/Annual.cshtml [hunk 9]
```diff
@@ -13,35 +13,43 @@
 <p>
```
[the remark this answers is not in the document]
Comment (mohamme15315) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/NoticestoMariners/Annual.cshtml:13]: As per Acceptance criteria "**Is the content table replaced by the error message in the screenshot including the reference?**" and also we confirmed with Brent that if error occurs on particular page then the static data should remain unchanged and the table should be replaced with the given error message.

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/NoticestoMariners/Cumulative.cshtml [hunk 10]
```diff
@@ -13,33 +13,40 @@
 <p>
```
[the remark this answers is not in the document]
Comment (mohamme15315) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/NoticestoMariners/Cumulative.cshtml:13]: As per Acceptance criteria "**Is the content table replaced by the error message in the screenshot including the reference?**" and also we confirmed with Brent that if error occurs on particular page then the static data should remain unchanged and the table should be replaced with the given error message.

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/NoticestoMariners/Annual.cshtml (hunk 9 above)
[the remark this answers is not in the document]
Comment (Chownder) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/NoticestoMariners/Annual.cshtml:13]: The wording of the ACs could allow for either way; if Brent is happy with it as is, then that's fine by me.

Still seems a little jarring to me personally, but I suppose the argument could be made that the static content is useful for reference whether or not there is an error in retrieving data.

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/NoticestoMariners/Cumulative.cshtml (hunk 10 above)
[the remark this answers is not in the document]
Comment (Chownder) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/NoticestoMariners/Cumulative.cshtml:13]: I'm happy with that explanation, as per the comment I made for Annual.

**Review by Chownder — APPROVED** [review]

**Pull request merged**